s2fft/utils/signal_generator.py

```diff
@@ -199,3 +214,60 @@ def generate_flmn(
                 rng, len_indices, var=2
             )
     return torch.from_numpy(flmn) if using_torch else flmn
+
+
+def generate_flm_from_spectra(
+    rng: np.random.Generator,
+    spectra: np.ndarray | jax.Array,
+) -> np.ndarray | jax.Array:
+    r"""
+    Generate a stack of random harmonic coefficients from power spectra.
+
+    The input power spectra must be a stack of shape *(K, K, L)* where
+    *K* is the number of fields to be sampled, and *L* is the harmonic
+    band-limit.
+
+    Args:
+        rng (Generator): Random number generator.
+
+        spectra (np.ndarray | jax.Array): Stack of angular power spectra.
+
+    Returns:
+        np.ndarray | jax.Array: A stack of random spherical harmonic
+        coefficients with the given power spectra.
+
+    """
+    # get the Array API namespace from spectra
+    xp = spectra.__array_namespace__()
```

Comment:
We are not requiring NumPy versions which guarantee Array API support, and this line is therefore causing test failures currently. Would either need to add `array-api-compat` package as a dependency and use that here or just avoid use of Array API for now.

Reply:
I see! If you are only targeting numpy and jax for the time being, then it's probably easiest to write a tiny custom `_get_array_namespace()` function instead of adding an extra dependency for this single function, and ignoring actual Array API compatibility (everything used here exists in both jax and numpy natively).

In any case, I'd rather not code this up as two identical functions that only differ in `np`/`jnp`.

Reply:
@matt-graham What approach would you recommend for now?

Reply:
I think @ntessore's current solution of adding a placeholder `_get_array_namespace` function which just returns the `numpy` or `jax.numpy` namespaces depending on the type of the array argument works well as a solution for now. In the longer term we probably do want to consider using the Array API more generally to avoid some of the duplication across JAX and NumPy implementations of functions Nicolas is trying to avoid here, and I suspect we will want to introduce `array-api-compat` as a dependency at that point to keep NumPy / JAX version requirements loose, but agree that adding this as a dependency for a single function is overkill here!